projects/plugins/jetpack/modules/sharedaddy/sharing.php

```diff
@@ -122,26 +124,26 @@ public function process_requests() {
 
 	/**
 	 * Register Sharing settings menu page.
+	 *
+	 * We only need to register it on self-hosted sites, in offline mode.
 	 */
 	public function subscription_menu() {
-		if ( ! defined( 'IS_WPCOM' ) || ! IS_WPCOM ) {
-			$active = Jetpack::get_active_modules();
-			if (
-				! in_array( 'publicize', $active, true )
-				&& ! current_user_can( 'manage_options' )
-			) {
-				return;
-			}
+		// Hide the menu on the wpcom platform.
+		if ( ( new Host() )->is_wpcom_platform() ) {
```

Comment:
Is this necessary? I believe we should be matching JP self hosted behavior but don't fully understand what I'm looking at here.